optimade/server/routers/utils.py

```diff
@@ -319,3 +324,25 @@ def get_single_entry(
         ),
         included=included,
     )
+
+
+def check_params(
+    request: Request, params: Union[EntryListingQueryParams, SingleEntryQueryParams]
+):
+    for param in request.query_params.keys():
+        if not hasattr(params, param):
+            split_param = param.split("_")
+            if param.startswith("_") and len(split_param) > 2:
```

Comment:
Personally I would suggest looping over all the parameters and accumulating all the errors and warnings before emitting them. 

It would probably be fine to just raise one error for all unrecognised fields, e.g.

"The query parameter(s) ['fitler', '_exmpl_test'] are not recognised by this entrypoint." 

Something like:

```python
for param in request.query_params.keys():
    errors = []
    warnings = []
    if not hasattr(params, param):
        split_param = params.split("_")
        if param.startswith("_") and len(split_param) > 2:
            prefix = split_param[1]
            if prefix in BaseResourceMapper.SUPPORTED_PREFIXES:
                errors.append(param)
            elif prefix not in BaseResourceMapper.KNOWN_PROVIDER_PREFIXES:
                warnings.append(param)
        else:
            errors.append(param)

    if warnings:
         warn(f"The query parameter(s) '{warnings}' are unrecognised and have been ignored.")
 
    if errors:
         raise BadRequest(f"The query parameter(s) '{errors}' are not recognised by this endpoint.")
```